pyrefly/lib/alt/expr.rs

```diff
@@ -705,6 +729,33 @@ impl<'a, Ans: LookupAnswer> AnswersSolver<'a, Ans> {
         }
     }
 
+    /// Check if a protocol is a data protocol (has non-method members)
+    fn is_data_protocol(&self, cls: &Class) -> bool {
+        // A data protocol has at least one non-method member
+        // For simplicity, we consider any annotated field that's not a method to be a data member
+        // This is a conservative approach that works for most practical cases
+        for field_name in cls.fields() {
+            if cls.is_field_annotated(field_name) {
+                // Use the class type to access the field 
+                let class_type = cls.as_class_type();
+                let ty = self.type_of_attr_get(
+                    &class_type.to_type(),
+                    field_name,
+                    ruff_text_size::TextRange::default(),
```

Comment:
If you instead add a `range` parameter and pass it in when `is_data_protocol` is called (line 695), you can use that here instead of a default.